vocabularies/Common.xml

```diff
@@ -1454,6 +1454,132 @@ If the request succeeds the response will contain a JSON body of `Content-Type:
         <Annotation Term="Core.Description" String="Base URL for WebSocket connections" />
       </Term>
 
+      <Term Name="RecommendationsFunction" Type="Common.QualifiedName" Nullable="false" AppliesTo="EntityType">
+        <Annotation Term="Common.Experimental" />
+        <Annotation Term="Core.Description" String="Qualified name of a function that computes AI-based recommendations for entities of the annotated type (see [this template](#Template_RecommendationsFunction) and [this alternative](#Template_RecommendationsFunction2))" />
```

Comment:
Short-term, we will have a function like the second alternative, but call it synchronously.
Mid-term, I prefer the first alternative, which makes the recommendations retrieval asynchronous.
Long-term, we can use `Prefer: respond-async` for _one_ request in a $batch, but only when Gateway supports that.

Reply:
I would skip the mid-term thingy because it is structurally different.

Just go for sync single bound function now and then evolve to async single bound function.

Reply:
We might need the mid-term thingy for 2408, if Gateway cannot implement the `Prefer: respond-async` in a $batch soon enough.

Reply:
The mid-term thingy is a detour with a different choreography. The cost of teaching this workaround to clients and then teaching them to _no longer use it_ should not be underestimated.

Better wait for the real async solution.

Reply:
I have removed any mention of the workaround from the vocabulary.